components/match2/wikis/ageofempires/match_group_input_custom.lua

```diff
@@ -12,520 +12,342 @@ local Game = require('Module:Game')
+
+		MatchGroupInputUtil.getCommonTournamentVars(map, match)
+
+		map.participants = CustomMatchGroupInput.processPlayerMapData(map, opponents)
+
+		map.finished = MatchGroupInputUtil.mapIsFinished(map)
+		local opponentInfo = Array.map(opponents, function(_, opponentIndex)
+			local score, status = MatchGroupInputUtil.computeOpponentScore({
+				walkover = map.walkover,
+				winner = map.winner,
+				opponentIndex = opponentIndex,
+				score = map['score' .. opponentIndex],
+			}, CustomMatchGroupInput.calculateMapScore(map.winner, map.finished))
+			return {score = score, status = status}
+		end)
+
+		map.scores = Array.map(opponentInfo, Operator.property('score'))
+		if map.finished then
+			map.resulttype = MatchGroupInputUtil.getResultType(winnerInput, finishedInput, opponentInfo)
+			map.walkover = MatchGroupInputUtil.getWalkover(map.resulttype, opponentInfo)
+			map.winner = MatchGroupInputUtil.getWinner(map.resulttype, winnerInput, opponentInfo)
 		end
+
+		table.insert(maps, map)
+		match[key] = nil
 	end
 
-	return match
+	return maps
 end
 
----@param record table
----@param timestamp number
-function CustomMatchGroupInput.processOpponent(record, timestamp)
-	local opponent = Opponent.readOpponentArgs(record) or Opponent.blank()
+---@param bestofInput string|integer?
+---@return integer?
+function CustomMatchGroupInput.getBestOf(bestofInput)
+	local bestof = tonumber(bestofInput) or tonumber(Variables.varDefault('bestof'))
 
-	-- Convert byes to literals
-	if Opponent.isBye(opponent) then
-		opponent = {type = Opponent.literal, name = 'BYE'}
+	if bestof then
+		Variables.varDefine('bestof', bestof)
 	end
 
-	---@type number|string
-	local teamTemplateDate = timestamp
-	-- If date is default date, resolve using tournament dates instead
-	-- default date indicates that the match is missing a date
-	-- In order to get correct child team template, we will use an approximately date and not the default date
-	if teamTemplateDate == DateExt.defaultTimestamp then
-		teamTemplateDate = DateExt.getContextualDateOrNow()
-	end
-
-	Opponent.resolve(opponent, teamTemplateDate, {syncPlayer = true})
-	MatchGroupInput.mergeRecordWithOpponent(record, opponent)
+	return bestof
 end
 
 ---@param match table
----@param mapIndex integer
-function CustomMatchGroupInput._mapInput(match, mapIndex)
-	local map = Json.parseIfString(match['map' .. mapIndex])
-	if String.isNotEmpty(map.map) and map.map ~= 'TBD' then
-		if Logic.isNotEmpty(match.mapsInfo) then
-			local info = Array.find(match.mapsInfo, function(m)
-				return m.name == map.map or m.link == map.map
-			end)
-			if info then
-				map.map = info.link
-				map.mapDisplayName = info.name
-			end
-		else
-			map.mapDisplayName = map.map
-			map.map = mw.ext.TeamLiquidIntegration.resolve_redirect(map.map or '')
-		end
-	end
+---@return string?, table?
+function CustomMatchGroupInput._getMapsAndGame(match)
+	local mapsInfo = Json.parse(Variables.varDefault('tournament_maps'))
```

Comment:
```suggestion
	local mapsInfo = Json.parseIfString(Variables.varDefault('tournament_maps'))
```
to prevent unnecessary error log output in case of section preview